build.gradle

```
@@ -1,83 +1,261 @@
+        classpath "net.minecraftforge.gradle:ForgeGradle:1.2-SNAPSHOT"
+        classpath "com.github.jengelman.gradle.plugins:shadow:4.0.3"
     }
 }
 
-apply plugin: 'forge'
+apply plugin: "java"
+apply plugin: "idea"
+apply plugin: "forge"
+apply plugin: "com.github.johnrengelman.shadow"
+apply plugin: "maven-publish"
 
-version = "2.0.0-BETA.2"
-group= "vulcanforge.pvp-mode-mod" // http://maven.apache.org/guides/mini/guide-naming-conventions.html
-archivesBaseName = "pvp-mode"
+ext.config = parseConfig(file("build.properties"))
+ext.dependenciesConfig = parseConfig(file("dependencies.properties"))
+ext.secrets = parseConfig(file("secrets.properties"))
 
-srcCompat = JavaVersion.VERSION_1_8
-targetCompat = JavaVersion.VERSION_1_8
+version = config.version
+group = config.group
+archivesBaseName = config.archivesBaseName
 
-minecraft {
-    version = "1.7.10-10.13.4.1614-1.7.10"
-    runDir = "eclipse"
+// Need this here so Eclipse tasks generates correctly
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
+compileJava {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+// This makes IDEA recognize the assets folder
+idea {
+    module {
+        inheritOutputDirs = true
+    }
+}
+
+repositories {
+    maven {
+        name = "amuzil"
+        url = "http://repo.amuzil.com/repository/public/"
+    }
 }
 
 configurations {
-    exports
+    shadow
+    compile.extendsFrom provided
+    provided.extendsFrom shadow
 }
 
+task useDeobfDependencies {
+    dependenciesConfig.obfuscated = false
+}
+
+tasks.runClient.dependsOn tasks.useDeobfDependencies
+tasks.runServer.dependsOn tasks.useDeobfDependencies
+
 dependencies {
-    exports 'io.github.lukehutch:fast-classpath-scanner:2.21'
-    configurations.compile.extendsFrom(configurations.exports)
+    provided fileTree(dir: dependenciesConfig.libraries + (dependenciesConfig.obfuscated ? "" : "Deobf"), include: ["*.jar"])
+
+    // Forge Mods
+    if (dependenciesConfig.obfuscated) {
+        provided "com.enderio.core:endercore:0.2.0-beta.39"
+        provided "com.timgodreuil.suffixforge:suffixforge:2.4.0"
+        provided "crazypants.enderio:enderio:2.3.0-beta.429"
+        provided "cyano.lootable:lootable-bodies:1.3.7"
+        provided "lotr:lotr:1.34.3"
+        provided "siege:siegemode:1.1.0"
+    } else {
+        provided "com.enderio.core:endercore:0.2.0-beta.39-deobf"
+        provided "com.timgodreuil.suffixforge:suffixforge:2.4.0-deobf"
+        provided "crazypants.enderio:enderio:2.3.0-beta.429-deobf"
+        provided "cyano.lootable:lootable-bodies:1.3.7-deobf"
+        provided "lotr:lotr:1.34.3-deobf"
+        provided "siege:siegemode:1.1.0-deobf"
+    }
+
+    // Bukkit Stuff
+    provided "me.raum.deathcraft:deathcraft:1.1.12-alpha"
+    provided "net.citizensnpcs:citizens:2.0.13-alpha.1"
+    provided "org.bukkit:craftbukkit-remapped:1.7.10"
+
+
```

Please replace that empty line with a comment.
```suggestion
// Dependencies that will be included ("shadowed") into the built JAR
```